lib/services/plugins-service.ts

```diff
@@ -77,12 +78,24 @@ export class PluginsService implements IPluginsService {
 					shelljs.cp("-R", pluginData.fullPath, pluginDestinationPath);
 				
 					let pluginPlatformsFolderPath = path.join(pluginDestinationPath, pluginData.name, "platforms", platform);
-					let pluginConfigurationFilePath = path.join(pluginPlatformsFolderPath, platformData.configurationFileName);					
+					let pluginConfigurationFilePath = path.join(pluginPlatformsFolderPath, platformData.configurationFileName);
+					let configurationFilePath = platformData.configurationFilePath;
+					
 					if(this.$fs.exists(pluginConfigurationFilePath).wait()) {
+						// Validate plugin configuration file
 						let pluginConfigurationFileContent = this.$fs.readText(pluginConfigurationFilePath).wait();
-						let configurationFileContent = this.$fs.readText(platformData.configurationFilePath).wait();
-						let resultXml = this.mergeXml(pluginConfigurationFileContent, configurationFileContent).wait();
-						this.$fs.writeFile(platformData.configurationFilePath, resultXml).wait();	
+						let doc = new DOMParser({
+							locator: {},
+							errorHandler: (level: any, msg: string) => {
+								this.$errors.fail(`Invalid xml file ${pluginConfigurationFilePath}. Additional technical information: ${msg}.` )
+							}
+						});
+						doc.parseFromString(pluginConfigurationFileContent, 'text/xml');
+						
+						// Merge xml
+						let configurationFileContent = this.$fs.readText(configurationFilePath).wait();
+						let resultXml = this.mergeXml(configurationFileContent, pluginConfigurationFileContent, platformData.mergeXmlConfig || []).wait();
```

Comment:
shouldn't you verify the resultXml as well? This way we will not save incorrect xml, which the user will have to modify manually after that.